zipline/protocol.py

```diff
@@ -194,6 +200,87 @@ def current_portfolio_weights(self):
         return position_values / self.portfolio_value
 
 
+class AlgorithmPortfolio(Portfolio):
+
+    def __init__(self, data_portal, benchmark=None):
+        super(AlgorithmPortfolio, self).__init__()
+        self.data_portal = data_portal
+        self.benchmark = benchmark
+        self.current_date = None
+
+    @property
+    def expected_shortfall(self):
+        """
+        Function for computing expected shortfall (also known as CVaR, or
+        Conditional Value at Risk) for the portfolio according to the assets
+        currently held and their respective weight in the portfolio.
+
+        This function requires a data portal in order to retrieve price
+        histories of the assets in the portfolio.
+        """
+        data_portal = self.data_portal
+        current_date = self.current_date
+
+        # If we do not even have a year's worth of data to look back on
```

Comment:
I don't think it's correct for us to be basing this decision on a relative offset from our start date.  A cardinal rule in Zipline is that simulation behavior shouldn't change depending on when my backtest starts and ends, but in this case, asking for the expected shortfall of a portfolio at time T depends on the relative offset of T from the start of the backtest.

I think the right behavior here is to fetch prices for all relevant assets and then filter those assets based on how much missing data they have.  One of the interesting questions is whether/how we want to redistribute the weights associated with the assets that get filtered out.  For example, if I have a portfolio of three assets, and one of them just IPOed, it's probably not right to just ignore 33% of the portfolio.  One option might be to redistribute the removed weight across other assets (though you'd probably want to treat longs and shorts separately).